lib/fastlane/plugin/firebase_app_distribution/helper/firebase_app_distribution_auth_client.rb

```diff
@@ -6,26 +6,42 @@ module FirebaseAppDistributionAuthClient
       TOKEN_CREDENTIAL_URI = "https://oauth2.googleapis.com/token"
 
       def fetch_auth_token(google_service_path)
+        if ENV["XDG_CONFIG_HOME"]
+          config_path = File.expand_path("configstore/firebase-tools.json", ENV["XDG_CONFIG_HOME"])
+        else
+          config_path = File.expand_path(".config/configstore/firebase-tools.json", "~")
+        end
         if !google_service_path.nil? && !google_service_path.empty?
           service_account(google_service_path)
         elsif ENV["FIREBASE_TOKEN"]
-          firebase_token
+          firebase_token(ENV["FIREBASE_TOKEN"])
         elsif ENV["GOOGLE_APPLICATION_CREDENTIALS"]
           service_account(ENV["GOOGLE_APPLICATION_CREDENTIALS"])
+        elsif File.exist?(config_path)
+          begin
+            refresh_token = JSON.parse(File.read(config_path))['tokens']['refresh_token']
+            if refresh_token.nil? || refresh_token.empty?
+              UI.crash!(ErrorMessage::MISSING_REFRESH_TOKEN)
```

Comment:
If we can't find the refresh token in the file I think that's _fine_ and we shouldn't crash. Instead we should fall back to the `UI.crash!(ErrorMessage::MISSING_CREDENTIALS)` since we basically were unable to find any valid auth mechanism.